### ISSUE TITLE
[pimatic-log-reader]

### ISSUE THREAD

**Issue by Ruba5913** [pimatic/pimatic#749]
I use LogWatcher to log a file pvlogtemp from eversolar-monitor.
But the graphics behave in a strange way and many times the pimatic webserver is frozen.
On the page zonnepanelen no acronyms are shown.
The amount of W and Wh stays always at the same amount until he is reading the log file than it acts like a counter!
Pimatic and plugins are up to date.
I made some screenshots.

[logwatcher.docx](https://github.com/pimatic/pimatic/files/37949/logwatcher.docx)
[pvlogtemp.docx](https://github.com/pimatic/pimatic/files/37951/pvlogtemp.docx)

My device code:

```
....
{
      "id": "Solar-energy",
      "name": "Solar-inverter",
      "class": "LogWatcher",
      "file": "/home/pi/eversolar-monitor/pvlogtemp",
      "xAttributeOptions": [
        {
          "name": "current power",
          "displaySparkline": false
        },
        {
          "name": "total power today",
          "displaySparkline": false
        }
      ],
      "attributes": [
        {
          "name": "current power",
          "type": "number",
          "unit": "W",
          "discrete": true,
          "acronym": "cp"
        },
        {
          "name": "total power today",
          "type": "number",
          "unit": "Wh",
          "discrete": true,
          "acronym": "tpt"
        }
      ],
      "lines": [
        {
          "match": "m[m++]=\"[0-9]+.[0-9]+.[0-9]+ [0-9]+:[0-9]+:[0-9]+|([0-9]+);[0-9]+;[0-9]+;[0-9]+;[0-9]+\"",
          "current power": "$1"
        },
        {
          "match": "m[m++]=\"[0-9]+.[0-9]+.[0-9]+ [0-9]+:[0-9]+:[0-9]+|[0-9]+;[0-9]+;([0-9]+);[0-9]+;[0-9]+\"",
          "total power today": "$1"
        }
      ]
    }
  ],
 ....
```
test data
```
m[mi++]="18.11.15 10:41:00|499;509;180;188;23"
m[mi++]="18.11.15 10:40:00|353;364;170;192;23"
m[mi++]="18.11.15 10:39:00|443;465;170;186;23"
m[mi++]="18.11.15 10:38:00|470;494;160;190;22"
m[mi++]="18.11.15 10:37:00|198;217;150;181;22"
m[mi++]="18.11.15 10:36:00|378;396;150;165;22"
m[mi++]="18.11.15 10:35:00|155;175;140;194;22"
m[mi++]="18.11.15 10:34:00|122;129;140;185;22"
m[mi++]="18.11.15 10:33:00|152;168;140;187;22"
m[mi++]="18.11.15 10:32:00|170;185;130;185;22"
m[mi++]="18.11.15 10:31:00|304;325;130;181;22"
m[mi++]="18.11.15 10:30:00|109;118;130;196;22"
m[mi++]="18.11.15 10:29:00|182;203;120;185;22"
m[mi++]="18.11.15 10:28:00|115;131;120;187;22"
m[mi++]="18.11.15 10:27:00|131;149;120;187;22"
m[mi++]="18.11.15 10:26:00|125;129;120;185;22"
m[mi++]="18.11.15 10:25:00|100;110;120;184;22"
```

Do I something wrong or is this a bug?
Anybody an idea ?

Thanks

**Comment by sweetpi** [comment]
1. The regex is wrong, try `"m[mi++]=\"[0-9]+.[0-9]+.[0-9]+ [0-9]+:[0-9]+:[0-9]+|([0-9]+);[0-9]+;[0-9]+;[0-9]+;[0-9]+\"`for the current power regex. 
2. Try attribute names without space
3. What do you mean by "acts like a counter"? The Log reader always uses the last matching line as value for the attribute. It does not sum up the values or count them.

**Comment by Ruba5913** [comment]
Hello 
@sweetpi

 thank you for your answer.

1. The regex what I uses gives no problems. When I delete the (") at the end like you suggested its        not accepted bij pimatic!
2. I removed the spaces but this has no effect.
3. I mean when I look to the solar inverter screen I see the amount of W and Wh changing rappidly when the log file is read.
When its ready the screen is showing the amount as before its starded to move and not the current amount.
Also in the graphic screen you see the W and Wh amount changing. (see blue arrow in the screenshot).
When the log file grows it takes a long time and screen freezes.
I attached a word file with screen shots ! 

[pimatic-log-reader.docx](https://github.com/pimatic/pimatic/files/48408/pimatic-log-reader.docx)

**Comment by Ruba5913** [comment]
I forgot to mention the acronyms aren't shown !!

**Comment by sweetpi** [comment]
> I mean when I look to the solar inverter screen I see the amount of W and Wh changing rappidly when the log file is read. When its ready the screen is showing the amount as before its starded to move and not the current amount. Also in the graphic screen you see the W and Wh amount changing. (see blue arrow in the screenshot). When the log file grows it takes a long time and screen freezes. I attached a word file with screen shots !

Are you using the most uptodate version of log-reader?

**Comment by Ruba5913** [comment]
Yes, I keep pimatic always up to date.
I using log-reader 0.8.7

**Comment by mwittig** [comment]
I am closing this now, as there has been no activity on this issue for a long time. The issue has been fixed recently and the fix will released with pimatic-log-reader@0.9.3

**Comment by mwittig** [comment]
I have reproduced the case and it works as expected with configuration and sample data included in the first post. I am closing this now
